lib/src/model/robotoff_nutrient_extraction_insight.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:openfoodfacts/src/model/nutrient.dart';
+import 'package:openfoodfacts/src/model/per_size.dart';
+
+import '../interface/json_object.dart';
+import 'robotoff_nutrient_extraction.dart';
+import 'robotoff_nutrient_extraction_annotation.dart';
+
+part 'robotoff_nutrient_extraction_insight.g.dart';
+
+@JsonSerializable()
+class RobotoffNutrientDataWrapper {
+  final Map<String, List<RobotoffNutrientEntity>>? entities;
+  final Map<String, RobotoffNutrientEntity>? nutrients;
+  final RobotoffNutrientAnnotation? annotation;
+  @JsonKey(name: 'was_updated')
+  final bool? wasUpdated;
+
+  const RobotoffNutrientDataWrapper({
+    this.entities,
+    this.nutrients,
+    this.annotation,
+    this.wasUpdated,
+  });
+
+  RobotoffNutrientEntity? getNutrientEntity({
+    required Nutrient nutrient,
+    required PerSize perSize,
+    String? entityKey,
+  }) {
+    String tag = nutrient.getOffTagPerSize(perSize);
+
+    if (entityKey != null) {
+      for (RobotoffNutrientEntity entity in entities?[entityKey] ?? []) {
+        if (entity.entity == tag) {
+          return entity;
+        }
+      }
+      return null;
+    }
+
+    return nutrients?[tag];
+  }
+
+  factory RobotoffNutrientDataWrapper.fromJson(Map<String, dynamic> json) =>
+      _$RobotoffNutrientDataWrapperFromJson(json);
+
+  Map<String, dynamic> toJson() => _$RobotoffNutrientDataWrapperToJson(this);
+}
+
+@JsonSerializable()
+class RobotoffNutrientExtractionInsight extends JsonObject {
+  @JsonKey(name: 'id')
+  String? insightId;
+  String? barcode;
+  // TODO: Figure out what this field is and what type it should be
+  // final String? type;
+  final RobotoffNutrientDataWrapper? data;
+  // @JsonKey(
+  //     fromJson: JsonHelper.timestampToDate, toJson: JsonHelper.dateToTimestamp)
+  // TODO ask server to send timestamp as milliseconds since epoch
+  // Current format is 2024-12-26T14:13:51.299483
+  final String? timestamp;
```

Comment:
We already manage those kinds of timestamp formats, I believe.
Please have a look at `JsonHelper`.